src/Integration.UnitTests/Notifications/SonarQubeNotificationServiceTests.cs

```diff
@@ -50,6 +50,8 @@ public class SonarQubeNotificationServiceTests
         [TestInitialize]
         public void TestInitialize()
         {
+            serverNotifications.Clear();
```

Comment:
This should not be needed. AFAIK before each test method is executed the runner creates a new instance of the test class. In other words, you cannot share state between test methods (easily) and you should not.

Reply:
Thanks for pointing it out, I wasn't aware! NUnit (which I used the most) executes everything on the same instance.